### ISSUE TITLE
Wrong export/import of Gutenberg post_content

### ISSUE THREAD

**Issue by manake** [WordPress/gutenberg#10684]
Exporting Gutenberg content via `get_posts` and then importing via `wp_insert_post` doesn't work. I also do `json_encode` and `json_decode`.

The problem is that `--` is converted to `\\u002d\\u002d`.

`"post_content": "<!-- wp:slug\/some-block {\"someAttributeOfThisBlock\":\"someValue\"} \/-->\n\n<!-- wp:slug\/some-other-block {\"someProp\":1234,\"someOtherProp\":\"xyz\",\"className\":\"x-y-z-a-b\\u002d\\u002dc\"} \/-->",`

**Comment by manake** [comment]
This is not solved in 4.0 RC-1:

https://github.com/WordPress/gutenberg/issues/9991
https://github.com/WordPress/gutenberg/issues/9929
https://github.com/WordPress/gutenberg/issues/7308
https://github.com/WordPress/gutenberg/issues/9423

I still have these two problems:
- 4.0 RC-1: `className` is not loaded back.
- 4.0 RC-1: Two dashes in 'className' are saved in such way that export/import doesn't work.

**Comment by designsimply** [comment]
Are the steps to reproduce and code snippet at https://github.com/WordPress/gutenberg/issues/9423#issuecomment-417262292 still relevant for testing for this case? I see that it was recommended in that discussion to use the forums or Slack for detailed technical questions and I also see you had further questions there about trying to double check whether it's really a bug or not.

**Comment by manake** [comment]
This is already solved for me: #9423 (I updated my above comment before you replied which says that this is no longer the case in Gutenberg 4.0 but it is the case in 3.9. I updated to 4.0 after I wrote this ticket and then I made a correction).

So this is the only problem that remains:
1. I can't do `get_posts()` and `wp_insert_post()` when two dashes are used next to one another.
2. To me it's not usable and it's a bug.

**Solution:**
On exporting the post double dashes must remain this: `x-y-z-a-b--c` and not this: `x-y-z-a-b\\u002d\\u002dc` (this is with `json_encode()` applied so the original is probably `x-y-z-a-b\u002d\u002dc`).
